sdk/typescript/telemetry/otlp.ts

```diff
@@ -0,0 +1,115 @@
+import { NodeSDK } from "@opentelemetry/sdk-node"
+import { SEMRESATTRS_SERVICE_NAME } from "@opentelemetry/semantic-conventions"
+import { Resource } from "@opentelemetry/resources"
+import {
+  BatchSpanProcessor,
+  NodeTracerProvider,
+} from "@opentelemetry/sdk-trace-node"
+import { OTLPTraceExporter } from "@opentelemetry/exporter-trace-otlp-grpc"
+import * as opentelemetry from "@opentelemetry/api"
+import { SpanStatusCode } from "@opentelemetry/api"
+import { credentials } from "@grpc/grpc-js"
+
+// Initialiaze the OTLP exporter, it takes his configuration from
+// the environment variables prefixed by "OTLP_"
+const exporter = new OTLPTraceExporter({
+  credentials: credentials.createInsecure(),
+})
```

Comment:
We need to make sure all the telemetry stuff is only executed if there's any `OTEL_` env variables set. Something like this:

https://github.com/dagger/dagger/blob/0aba66c0ddc170667b8c87f030e6a6eb38b8c1a2/sdk/go/telemetry/init.go#L24-L31

`dagger run` sets `TRACEPARENT`, which should still be propagated to request headers if it exists, but it doesn't set any other `OTEL_` env vars so we should skip configuring the exporter and everything in that case.